java/client/src/main/java/glide/api/commands/GenericBaseCommands.java

```diff
@@ -371,4 +371,22 @@ CompletableFuture<Boolean> pexpireAt(
      * }</pre>
      */
     CompletableFuture<String> type(String key);
+
+    /**
+     * Returns the internal encoding for the Redis object stored at <code>key</code>.
+     *
+     * @see <a href="https://redis.io/commands/object-encoding/">redis.io</a> for details.
+     * @param key The <code>key</code> of the object to get the internal encoding of.
+     * @return If <code>key</code> exists, returns the internal encoding of the object stored at
+     *     <code>key</code> as a <code>String</code>. Otherwise, return <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * String encoding = client.objectEncoding("key").get();
+     * assert encoding.equals("encoding");
```

Comment:
Maybe show real sample?
```suggestion
     * String encoding = client.objectEncoding("my_hash").get();
     * assert encoding.equals("listpack");
```